cmd/webhook.go

```diff
@@ -233,11 +233,16 @@ func createPatch(pod *corev1.Pod, sidecarConfigTemplate *Config, annotations map
 			// Should deep copy because things change
 			tempSidecarConfig, _ := deepcopy.Anything(sidecarConfigTemplate)
 			sidecarConfig := tempSidecarConfig.(*Config)
-			bucketName := string(secretList.Items[sec].Data["S3_BUCKET"])
+
+			// bucket might be a full path with shares, meaning with slashes
+			bucketMount := string(secretList.Items[sec].Data["S3_BUCKET"])
+			// clean the bucket name in case of deep path(like "path1/path2")
+			bucketName := strings.Replace(bucketMount, "/", "-", -1)
```

Comment:
Not sure this is correct.
If I understand the code correctly, the secret value "S3_bucket" will have the full path from the share `x_share/y_dir/z_dir/1_dir`

lets use that example, if we just replace for `bucketname` I assume it should be `x_share`

After this line runs, `bucketname` = `x_share-y_dir-z_dir-1_dir`

Container Names must be no longer than 63 characters, must start and end with a lowercase letter or number, and may contain lowercase letters, numbers, and hyphens.

Expressed as a regular expression:
`[a-z0-9]([-a-z0-9]*[a-z0-9])?`

Reply:
So you are suggesting we just take the first part of the string for the `bucketname`? So in that example, `bucketname` would just be `x_share`? 
Because my main issue with that is that it would cause duplicate problems if a user would have both `x_share/y_dir` and `x_share/z_dir`.
You do bring up a very good point about the length of the container name though and the regex. Although the `S3_BUCKET` value is curently something that has been manually created by us, that won't always be the case(hopefully), and this code currently doesn't have much in terms of validation for the secret values.
So it is not a bad idea to add those validations (the length limit and the regex). And the duplicate concern is not too hard to bypass as well. This code is running in a loop, just shove the loop interation in the name and it should solve it (so it would resolve as something like `$FILER-$SHARE-X-bucket-container`, where X is the loop interation number)

We might have be able to auto-correct the length, but if the bucketname resolves in something that doesnt match that regex, I'm not sure if we can auto-fix that easily